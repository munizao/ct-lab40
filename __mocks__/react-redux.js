jest.mock('react-redux', () => {
  return {
    __esModule: true,
    useDispatch: jest.fn(() => () => {}),
    useSelector: jest.fn((selector) => selector({ user: {}, repos: {} })),
  };
});
