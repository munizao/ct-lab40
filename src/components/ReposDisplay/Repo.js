import React from 'react';
import PropTypes from 'prop-types';

const Repo = ({ repo }) => {
  return (
    <li>
      <p>{repo.name}</p>
      <p>Issues: {repo.open_issues}</p>
      <p>Forks: {repo.forks}</p>
      <p>Stars: {repo.stargazers_count}</p>
      <p>Created: {Date(repo.created_at)}</p>
      <p>Last Update: {Date(repo.updated_at)}</p>
      <p>Description: {repo.description}</p>
    </li>
  );
};

Repo.propTypes = {
  repo: PropTypes.shape({
    name: PropTypes.string.isRequired,
    open_issues: PropTypes.number.isRequired,
    forks: PropTypes.number.isRequired,
    stargazers_count: PropTypes.string.isRequired,
    created_at: PropTypes.string.isRequired,
    updated_at: PropTypes.string.isRequired,
    description: PropTypes.string.isRequired
  })
};

export default Repo;
