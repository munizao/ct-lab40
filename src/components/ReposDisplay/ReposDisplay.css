.ReposDisplay {
  width: 75vw;
}