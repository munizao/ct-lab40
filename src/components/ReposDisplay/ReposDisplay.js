import React, { useEffect } from 'react';
import { useSelector, useDispatch } from 'react-redux';

import { selectRepos, selectLoading } from '../../redux/selectors/repos-selector';
import { selectUser } from '../../redux/selectors/user-selector';
import { fetchRepos } from '../../redux/actions/actions';
import Repo from './Repo';
import styles from './ReposDisplay.css';

const ReposDisplay = () => {
  const reposInfo = useSelector(selectRepos);
  const loading = useSelector(selectLoading);
  const user = useSelector(selectUser);
  const dispatch = useDispatch();

  useEffect(() => {
    if(user) {
      dispatch(fetchRepos(user.login));
    }
  }, [user && user.login]);

  return (
    <div className={styles.ReposDisplay}>
      {loading && 'Loading...'}
      {
        reposInfo &&
          <div>
            {reposInfo.map((repo) => <Repo key={repo.id} repo={repo} />)}
          </div>
      }
    </div>
  );
};

export default ReposDisplay;
