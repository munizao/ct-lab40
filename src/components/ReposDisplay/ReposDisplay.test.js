import React from 'react';
import 'react-redux';
import { shallow } from 'enzyme';
import ReposDisplay from './ReposDisplay';

describe('ReposDisplay component', () => {
  it('renders ReposDisplay', () => {
    const wrapper = shallow(<ReposDisplay />);
    expect(wrapper).toMatchSnapshot();
  });
});
