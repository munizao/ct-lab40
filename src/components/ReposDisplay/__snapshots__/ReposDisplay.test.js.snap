// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReposDisplay component renders ReposDisplay 1`] = `
<div
  className="ReposDisplay"
/>
`;
